h5netcdf/core.py

```diff
@@ -724,6 +748,26 @@ def _create_child_variable(
         else:
             self._root._check_valid_netcdf_dtype(dtype)
 
+        # compound type handling
+        if np.dtype(dtype).kind == "c":
+            if not isinstance(dtype, CompoundType):
+                # todo check compound types for
+                #  existing complex types which may be used
+                itemsize = np.dtype(dtype).itemsize
+                try:
+                    width = {8: "FLOAT", 16: "DOUBLE"}[itemsize]
+                except KeyError as e:
+                    raise TypeError(
+                        "Currently only 'complex64' and 'complex128' dtypes are allowed."
+                    ) from e
+                dname = f"_PFNC_{width}_COMPLEX_TYPE"
+                dtype = self._all_cmptypes.get(
+                    dname, self.create_cmptype(np.dtype(dtype), dname)
+                )
+            # use committed type
+            # that will actually reference the named type in the dataset
+            dtype = dtype._h5ds
+
```

Comment:
This actually references the committed type within the dataset. I've used `h5ls` to check this. This tool was recommended at hdf5forum. For some reason I can't retrieve the `name` of the committed type from the dataset datatype. I've struggled to retrieve this from the plain c code over at hdf5 but that did not work out. 

There is one solution to check the dataset type with any committed type for equality, but this seems quite clunky. I'll implement that solution to move forward here.

Reply:
Just before giving up, I've found another way of retrieving the name of the linked type directly from the dataset (here `datac2`):

```python
ds._root._h5py.h5i.get_name(datac2._h5ds.id.get_type()).decode()
```

Getting back to this the next days. @shoyer Sorry, but this is more involved than I anticipated.